gen/field.go

```diff
@@ -81,57 +86,95 @@ func (f fieldGroupGenerator) DefineStruct(g Generator) error {
 }
 
 func (f fieldGroupGenerator) ToWire(g Generator) error {
-	// TODO(abg): Default values
 	return g.DeclareFromTemplate(
 		`
 		<$wire := import "github.com/thriftrw/thriftrw-go/wire">
 
 		<$v := newVar "v">
-		func (<$v> *<.Name>) ToWire() <$wire>.Value {
+		func (<$v> *<.Name>) ToWire() (<$wire>.Value, error) {
     		<$fields := newVar "fields">
     		<$i := newVar "i">
-			// TODO check if required fields that are reference types are nil
+			<$wVal := newVar "w">
 
-			var <$fields> [<len .Fields>]<$wire>.Field
-			<$i> := 0
+			<if and .IsUnion (len .Fields)>
+				<$fmt := import "fmt">
+				<$count := newVar "count">
+				<$count> := 0
+				<range .Fields>
+					if <$v>.<goCase .Name> != nil { <$count>++ }
```

Comment:
any reason why this validation is done separately, instead of just using `<$i>` after all the field validation? While required validation is done inline